scorep/instrumenters/scorep_profile.py

```diff
@@ -23,146 +22,28 @@ def _unsetprofile():
         threading.setprofile(None)
 
 
-class ScorepProfile(base_instrumenter.BaseInstrumenter):
-    def __init__(self, scorep_bindings, enable_instrumenter=True):
-        """
-        @param enable_instrumenter true if the tracing shall be initialised.
-            Please note, that it is still possible to enable the tracing later using register()
-        """
-        global global_instrumenter
-        global_instrumenter = self
-        self.tracer_registered = False
-
-        self.scorep_bindings = scorep_bindings
-        self.globaltrace = self.globaltrace_lt
-        self.enable_instrumenter = enable_instrumenter
-
-    def register(self):
-        self.tracer_registered = True
-        _setprofile(self.globaltrace)
+class ScorepProfile(ScorepInstrumenterBase):
+    def _enable_instrumenter(self):
+        _setprofile(self._globaltrace)
 
-    def unregister(self):
+    def _disable_instrumenter(self):
         _unsetprofile()
-        self.tracer_registered = False
-
-    def get_registered(self):
-        return self.tracer_registered
-
-    def run(self, cmd):
-        self.runctx(cmd)
-
-    def runctx(self, cmd, globals=None, locals=None):
-        if globals is None:
-            globals = {}
-        if locals is None:
-            locals = {}
-        if self.enable_instrumenter:
-            self.register()
-        try:
-            exec(cmd, globals, locals)
-        finally:
-            self.unregister()
 
-    def runfunc(self, func, *args, **kw):
-        result = None
-        if self.enable_instrumenter:
-            self.register()
-        try:
-            result = func(*args, **kw)
-        finally:
-            self.unregister()
-        return result
-
-    def globaltrace_lt(self, frame, why, arg):
+    def _globaltrace(self, frame, why, arg):
         """Handler for call events.
 
         If the code block being entered is to be ignored, returns `None',
         else returns self.localtrace.
         """
         if why == 'call':
             code = frame.f_code
-            modulename = base_instrumenter.get_module_name(frame)
+            modulename = get_module_name(frame)
             if not code.co_name == "_unsetprofile" and not modulename[:6] == "scorep":
-                full_file_name = base_instrumenter.get_file_name(frame)
+                full_file_name = get_file_name(frame)
                 line_number = code.co_firstlineno
-                self.scorep_bindings.region_begin(
-                    modulename, code.co_name, full_file_name, line_number)
-            return
+                self.region_begin(modulename, code.co_name, full_file_name, line_number)
```

Comment:
As you always talk about overhead: This is another indirection, which causes another function call and therefore more overhead 😉 .

Reply:
I know. I decided on less code duplication here in case the `region_begin` needs changing and to not use the private member and cleaner code.

Reply:
Changed: d281479